Yoast/wordpress-seo
# Make progress bars work in React snippet editor

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

## Test instructions

This PR can be tested by following these steps:

Please look at https://github.com/Yoast/YoastSEO.js/pull/1485 and https://github.com/Yoast/yoast-components/pull/493 first.

* Please test in the classic editor.
* Use define( 'YOAST_FEATURE_SNIPPET_PREVIEW', true ); to see the react snippet editor (for example on line 46 in `class-metabox-formatter.php`, or in your `wp-config.php`).
* Link the branches from https://github.com/Yoast/YoastSEO.js/pull/1485 and https://github.com/Yoast/yoast-components/pull/493.
   * Link `yoastseo.js` and `yoast-components` to `wordpress-seo`. 
   * Link `yoastseo` to `yoast-components` using `ln -s [path to your local yoastseo folder] node_modules/yoastseo` in `yoast-components` + remove `.babelrc` from yoastseo to avoid build errors. (It's ugly, I know)
   * Don't forget to run `yarn install` afterwards.

### Title progress bar
* Write some titles in the react snippet editor and see the progress bar reflecting the length, both by progress and color. 0-399 pixels = orange bar, 400-599 pixels = green bar, >600 pixels = red bar. This should work for both the mobile view as well as the desktop view.

### Metadescription progress bar
* Both in the desktop view and mobile view, write some meta descriptions in the react snippet editor and see the progress bar reflecting the length, both by progress and color (with a delay, see 'Known issues that don't block merging' below). 0-120 and >320 characters = orange bar, 121-320 characters = green bar.
* NB: in the 'old' snippet preview, the meta description turns red when it's too long. We don't want that anymore (as discussed with @jdevalk and @jono-alderson)

### Other
* Look if nothing else breaks, especially in the content analysis.

## Important for merging
- First merge https://github.com/Yoast/YoastSEO.js/pull/1485 and https://github.com/Yoast/yoast-components/pull/493 first.
- Release yoast-components and yoastseo.js (please discuss yoastseo.js release with team frontend).
- Change the versions of yoast-components and yoastseo.js in the package.json in this branch. Also commit the new yarn.lock.
- Merge this PR

## Quality assurance


Fixes #9588
